### PR TITLE
[DOC-538] guide for writing a multi-asset integration

### PR DESCRIPTION
**Pull request by cmpadden**
## Summary & Motivation

Migration of guide on writing a multi-asset integration to the new docs.

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

**Comment by cmpadden** [comment]
Including you as a reviewer on this one, @neverett , but it's low prio.

Eventually we'll want to have guides on how to write integrations with Dagster, and this was a port of one that I wrote in the legacy docs.

**Comment by neverett** [comment]
Cross-linking https://github.com/dagster-io/dagster/pull/26323, which reorganizes the integrations section to make space for narrative docs like this.

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+# Create a multi-asset integration
+
+When working in the Dagster ecosystem, you may have noticed that decorators are frequently used. For example, assets, jobs, and ops use decorators. If you have a service that produces many assets, it's possible to define it as a multi-asset decorator-offering a consistent and intuitive developer experience to existing Dagster APIs.
+
+In the context of Dagster, decorators are helpful because they often wrap some form of processing. For example, when writing an asset, you define your processing code and then annotate the function with the `asset` decorator /> decorator. Then, the internal Dagster code can register the asset, assign metadata, pass in context data, or perform any other variety of operations that are required to integrate your asset code with the Dagster platform.
+
+In this guide, you'll learn how to develop a multi-asset integration for a hypothetical replication tool.
+
+---
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:14]:
```suggestion
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+title: Create a multi-asset integration
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:2]:
```suggestion
title: Creating a multi-asset integration
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+description: Create a decorator based multi-asset integration
+---
+
+# Create a multi-asset integration
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:6]:
```suggestion
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+---
+
+## Prerequisites
+
+To follow the steps in this guide, you'll need:
+
+- Familiarity with Dagster
+- An understanding of Python decorators—[Real Python's Primer on Python Decorators](https://realpython.com/primer-on-python-decorators/) is a fantastic introduction
+
+---
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:23]:
```suggestion
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+
+def replicate(replication_configuration_yaml: Path) -> Iterator[Mapping[str, Any]]:
+    data = yaml.safe_load(replication_configuration_yaml.read_text())
+    for table in data.get("tables"):
+        # < perform replication here, and get status >
+        yield {"table": table.get("name"), "status": "success"}
+```
+
+---
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:65]:
```suggestion
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+```
+
+There are a few limitations to this approach:
+
+- **We have not encapsulated the logic for replicating tables.** This means that users who use the `custom_replication_assets` decorator would be responsible for yielding asset materializations themselves.
+- **Users can't customize the attributes of the asset**.
+
+For the first limitation, we can resolve this by refactoring the code in the body of our asset function into a Dagster resource.
+
+---
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:146]:
```suggestion
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+@custom_replication_assets(
+    replication_project=replication_project,
+    name="my_custom_replication_assets",
+    group_name="replication",
+)
+def my_assets(replication_resource: ReplicationProject):
+    replication_resource.run(replication_project)
+```
+
+---
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:185]:
```suggestion
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+
+## Step 4: Using translators
+
+At the end of [Step 2](#step-2-implementation), we mentioned that end users were unable to customize asset attributes, like the asset key, generated by our decorator. Translator classes are the recommended way of defining this logic, and they provide users with the option to override the default methods used to convert a concept from your tool (for example, a table name) to the corresponding concept in Dagster (for example, asset key).
+
+To start, we will define a translator method to map the table specification to a Dagster asset key. **Note**: in a real world integration you will want to define methods for all common attributes like dependencies, group names, and metadata.
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:191]:
```suggestion
To start, we will define a translator method to map the table specification to a Dagster asset key.

:::note
In a real world integration, you will want to define methods for all common attributes like dependencies, group names, and metadata.
:::
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+
+
+@dataclass
+class ReplicationTranslator:
+    @public
+    def get_asset_key(self, table_definition: Mapping[str, str]) -> AssetKey:
+        return AssetKey(str(table_definition.get("name")))
+```
+
+Next, we'll update `custom_replication_assets` to use the translator when defining the `key` on the `AssetSpec`. **Note** that we took this opportunity to also include the replication project and translator instance on the `AssetSpec` metadata. This is a workaround that we tend to employ in this approach, as it makes it possible to define these objects once and then access them on the context of our asset.
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:206]:
```suggestion
Next, we'll update `custom_replication_assets` to use the translator when defining the `key` on the `AssetSpec`. 

:::note
Note that we took this opportunity to also include the replication project and translator instance on the `AssetSpec` metadata. This is a workaround that we tend to employ in this approach, as it makes it possible to define these objects once and then access them on the context of our asset.
:::
```

File: docs/docs-beta/docs/tutorial/multi-asset-integration.md [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+
+        results = replicate(Path(project.replication_configuration_yaml))
+        for table in results:
+            if table.get("status") == "SUCCESS":
+                yield AssetMaterialization(
+                    asset_key=translator.get_asset_key(table), metadata=table
+                )
+```
+
+---
```
Comment (neverett) [inline docs/docs-beta/docs/tutorial/multi-asset-integration.md:268]:
```suggestion
```

**Review by neverett — APPROVED** [review]
Left a few small editorial suggestions, otherwise this LGTM, thanks!

You'll also need to merge master in and move this doc to docs/integrations/guides -- you can delete the empty multi-asset-integration.md file that's already in there.

**Pull request merged**